### ISSUE TITLE
HF-cache delete guard fails open for every @dgxrun/ deploy (offers Delete on serving weights)

### ISSUE THREAD

**Issue by kreuzhofer** [kreuzhofer/dgx-manager#85]
## Problem

The HF-cache delete guard resolves recipe → HF-repo ids from the **sparkrun catalog only**. Every `@dgxrun/` deployment therefore fails to resolve, its HF repo is never matched, and the model list offers **Delete** for weights that are actively serving.

This is a **fail-open delete guard**. The module's own comment states the intended bias:

> *"This is the safe bias for a delete guard: a false 'in use' merely annoys; a false 'deletable' can rm the weights out from under a live container."*

It is currently failing in the dangerous direction.

## Reproduced 2026-09-04

`LibertAIDAI/GLM-5.3-Flash-NVFP4` was serving on dgx-spark-02 + dgx-spark-03 (deployment `cmtmtc64857kx2auheovbmrla`, `status: running`, answering on `/v1/chat/completions`) while `GET /api/hf-cache` reported:

```
LibertAIDAI/GLM-5.3-Flash-NVFP4   inUse=False  inUseBy=[]
Qwen/Qwen3.8-27B                  inUse=True   inUseBy=['qwen3.8-sql-500step-r2']
unsloth/Qwen3.8-27B-NVFP4         inUse=False  inUseBy=[]
```

so the dashboard shows **Delete** for GLM-5.3-Flash and **in use** for Qwen3.8.

## Cause

`packages/server/src/routes/hf-cache.ts:28-31`:

```ts
const recipeHfId = new Map<string, string>();
for (const r of agentHub.getRecipes()) {      // <-- sparkrun catalog ONLY
  if (r.model) recipeHfId.set(r.file, r.model);
}
```

The GLM deployment's stored `model.name` is the **recipe ref** `@dgxrun/zai-glm-5.3-flash-libertai-nvfp4-2x`, not an HF id (`modelName` is null). The recipe file declares `model: LibertAIDAI/GLM-5.3-Flash-NVFP4`, but that recipe lives in the **dgxrun** catalog, which is never consulted — so `recipeHfId.get(recipeFile)` misses and the HF id never enters `candidates`. `matchRepoToModels` then correctly finds no match, and `repoUsage` reports `inUse: false`.

Qwen3.8-27B only shows "in use" by accident of a different code path: line 43, `d.model.finetuneJob?.baseModel`, which the fine-tune deploy populates. Nothing to do with recipe resolution.

## Why every dgxrun deploy is affected

`@dgxrun/` is the runner used for **all multi-node deploys** and for everything in `recipes/dgxrun/` — currently 11 recipes including the GLM-5.2/5.3 family, Qwen3.8 (bf16, nvfp4, rtx) and Muse Glimmer. So the guard is effectively absent for the most expensive, hardest-to-rebuild weights on the fleet.

## Same root cause as [#44](https://github.com/kreuzhofer/dgx-manager/issues/44)

`docs/ROADMAP.md` already records this exact shape for the heterogeneous-cluster guard:

> *"it is wired to `agentHub.getRecipes()` — the sparkrun catalog — so every `@dgxrun/` deploy misses the lookup and skips the check entirely. The dgxrun catalog already parses and validates `arch`; it is simply never consulted."*

Two independent safety guards, both silently disabled for dgxrun deploys, both for the same reason. Worth fixing together and worth grepping for other `getRecipes()` callers that assume it is the whole catalog.

## Fix

`getDgxrunCatalog()` (`packages/server/src/deployments/dgxrun-catalog.ts:112`) already returns entries carrying `file` (`@dgxrun/<base>`) and `model?: string` — exactly the two fields needed. Merge it into `recipeHfId`:

```ts
for (const r of getDgxrunCatalog()) {
  if (r.model) recipeHfId.set(r.file, r.model);
}
```

Note the catalog is memoized; `refreshDgxrunCatalog()` exists for invalidation.

## Suggested test

Integration test: create a running deployment whose `config.recipeFile` is a `@dgxrun/` ref and whose stored model name is the recipe ref rather than an HF id, then assert the recipe's declared HF repo comes back `inUse: true` and that `DELETE` is refused with 409. The existing tests in `hf-cache.routes.test.ts` evidently cover only sparkrun-shaped deployments.

Found while working [#79](https://github.com/kreuzhofer/dgx-manager/issues/79); spotted by Daniel noticing GLM-5.3-Flash offered Delete while deployed.

**Comment by kreuzhofer** [comment]
Fixed in `648300e` on `qwen3.8-27b-and-staging-design` (together with #44 — same root cause).

`loadDeploymentUsage` now builds `recipeHfId` from **both** catalogs:

```ts
for (const r of [...getDgxrunCatalog(), ...agentHub.getRecipes()]) {
  if (r.model) recipeHfId.set(r.file, r.model);
}
```

matching the merge order `routes/recipes.ts:37` already uses. The `@dgxrun/` prefix means the two key spaces can't collide.

Tests added to `hf-cache.routes.test.ts`, both against a stub hub reporting an **empty** sparkrun catalog so the resolution has to come from the dgxrun one:
- `GET /api/hf-cache` → a running `@dgxrun/zai-glm-5.3-flash-libertai-nvfp4-2x` deploy (`model.name` = the recipe ref, `modelName` null) marks `LibertAIDAI/GLM-5.3-Flash-NVFP4` `inUse: true`, `inUseBy: ["glm-flash-prod"]`.
- `DELETE /api/hf-cache/shared?repoId=…` → 409, nothing sent to the agent.

Both returned 200-with-`inUse:false` / 202 on the old code before the fix.

### Other `getRecipes()` callers, as suggested

Grepped them all. One more real gap, not fixed here because it isn't a fail-open guard and wasn't in scope:

- **`deployments.ts:564`** — `recordRepoDeployment`'s `recipeModel`. A dgxrun deploy never records its HF repo id, so the durable `lastDeployedAt` stays empty for exactly the weights this issue is about. Display gap only; the in-use guard above is now sound either way. Worth its own issue.

Benign:
- `:331` (cluster sizing from `tensor_parallel`) — already reads `dgxrunRecipe?.defaults` first.
- `:415` and `:905` (`vramEstimate` / restart `gpuMemUtil`) — a dgxrun miss falls back to 0.85 rather than the recipe's declared value. Estimate accuracy, not admission: the actual VRAM check at `:469` does consult `dgxrunRecipe.defaults`.

Full suite green: 1387/1387.
